### PR TITLE
RFC: jump to numbered stackframe with hotkey

### PR DESCRIPTION
**Pull request by KristofferC**
This PR allows you to jump to a stackframe by entering its number (showed in the stacktrace from pr #19569) in the REPL and then pressing CTRL+Q. The keybinding is just something I took, it can be discussed. 

I did a small recording of how it plays out: https://asciinema.org/a/17k6tfwec6hm6q57nzlbns61g

### CODE REVIEW

File: base/replutil.jl [hunk 1]
```diff
@@ -567,14 +567,21 @@ function show_trace_entry(io, frame, n; prefix = " in ")
     n > 1 && print(io, " (repeats ", n, " times)")
 end
 
+# Contains file name and file number. Gets set when a backtrace
+# is showed. Used by the REPL to make it possible to open
```
Comment (tkelman) [inline base/replutil.jl]: shown

File: base/REPL.jl [hunk 2]
```diff
@@ -899,6 +899,26 @@ function setup_interface(repl::LineEditREPL; hascolor = repl.hascolor, extra_rep
...
         end,
+
+        # Open the editor at the location of a stackframe
+        # This is accessing a global variable that gets set in
+        # the show_backtrace function.
+        "^Q" => (s, o...) -> begin
+            linfos = Base.LAST_BACKTRACE_LINE_INFOS
+            str = String(take!(LineEdit.buffer(s)))
+            n = tryparse(Int, str)
+            isnull(n) && @goto writeback
```
Comment (KristofferC) [inline base/REPL.jl]: These `@goto` shenanigans are probably not worth it. :)

**Comment by KristofferC** [comment]
Added some news.

Note that it is currently possible to jump to REPL frames. This could be done by saving all the inputs in the REPL and than possibly creating a temp file that gets opened in the editor.

File: base/REPL.jl (hunk 2 above)
Comment (StefanKarpinski) [inline base/REPL.jl]: I'm fine with it. This sort of interactive input use case is actually one of the places where goto seems to actually be clearer and easier than anything else.

**Review by StefanKarpinski — APPROVED** [review]
I'm not 100% sold on the UI of this but it's certainly better than nothing, which is what we have now, so I'm all for trying it out and seeing how it goes.

File: NEWS.md [hunk 3]
```diff
@@ -83,6 +83,8 @@ Library improvements
 
   * The `chop` and `chomp` functions now return a `SubString` ([#18339]).
 
+  * Numbered stackframes printed in stacktraces can be opened in an editor by entering the corresponding number in the REPL and pressing `^Q`.
```
Comment (tkelman) [inline NEWS.md]: add PR reference

Comment (KristofferC) [inline NEWS.md]: done

Comment (KristofferC) [inline NEWS.md]: oh now i got a merge conflict.

Comment (KristofferC) [inline NEWS.md]: resolved it

Comment (tkelman) [inline NEWS.md]: don't have to run news-update right away as long as the reference is in the right format for it

**Comment by StefanKarpinski** [comment]
@tkelman, shall we merge this?

**Comment by tkelman** [comment]
with a squash and taking the `ci skip`s out of the commit messages, yep

**Pull request merged**